### DISCUSSION TITLE
Information about cache managing on Android and IOS

### DISCUSSION THREAD

**Discussion by SNWStaff** [invertase/react-native-firebase#4143] [Q&A] [answered]
Hello to firebase community !

I have a question about how cache is managed with firebase in some cases. I read that when you subscribe to data, after getting them back to your callback, while listener is active, data persist in local, and there is no much query to database done, while data stay the same. 
But what if I close my app then re-open to display data I previously displayed, did they get cached and firebase have not to make a new request ? Or do Firebase have to get them back from remote database ? And what about Firebase web with react, did the data are cached if there are not changed ?
I need answer to this questions because I want to know if it's better to save data into a global store (like redux) and access to local values only when I need to display, or is it better to subscribe to data only when I'm on my display page and unsubscribe when I leave ?

Sincerely
Sweet New World Team

**Comment by Salakar** [comment]
@SNWStaff which Firebase database engine are you using, is it Firestore?

**Comment by SNWStaff** [comment, marked as answer]
Hello back, sorry for inactivity I totally forget about this post. Yes I use Firestore, and need some infos to know when my app will send request, and when it will use firebase cache, to analyse and prevent potential cost if many users use it at the same time. Thanks :)

**Reply by mikehardy** [reply]
I'm sorry - I think the only way to move forward with this since it will involve costs on your side, is to put a network proxy in between the app and firebase, and test everything while watching for traffic to see what is cached and when, and how you can affect it. Note that you can do this easily (I think) now with the firestore emulator so that it is all local and does not involve SSL certificates etc
